### PR TITLE
added more detailed version support info for private cloud platforms

### PR DESCRIPTION
**Pull request by sttomm**
I added more information around version requirements for private cloud platforms. I only added information that was retrievable with a reasonable amount of effort.

For OSB we already mention the version we support.

For public cloud platforms i don't see what should be documented around this. Mentioning some kind of root access to the platform as a prerequisite doesn't make too much sense for me. If there is additional information known by the solutions flock that should be documented, please feel free to extend this PR accordingly.

### CODE REVIEW

File: docs/meshstack.cloudfoundry.index.md [hunk 1]
```diff
@@ -19,6 +19,24 @@ The tenant [replication](./meshcloud.tenant.md) ensures spaces and orgs are crea
...
+* /v2/space_quota_definitions
+* /v2/spaces
+* /v2/users
+* /v2/apps
+* /v2/private_domains
+* /v2/domains
+
+Integration with the following Cloud Foundry API Versions are additionally tested successfully:
+
+* 2.155.0
```
Comment (JohannesRudolph) [inline docs/meshstack.cloudfoundry.index.md:37]: d: maybe put this first like

meshcloud officially validates compatibility with the following API versions. In general all version that provide V2 API for...

File: docs/meshstack.openshift.index.md [hunk 2]
```diff
@@ -15,7 +15,12 @@ To enable integration with OpenShift, operators deploy and configure the meshSta
 
 ### OpenShift Versions
 
-meshStack currently supports OpenShift version 3.9+ as either Open-Source (OKD) or OpenShift Enterprise variants.
+meshStack currently supports OpenShift version 3.9+ and 4.x as either Open-Source (OKD) or OpenShift Enterprise variants.
+
+In general meshStack supports all OpenShift versions that can provide the resources listed for the required [service accounts](meshstack.openshift.index.md#meshstack-service-accounts). These resources are consumed with the following versions. So as long as the following endpoints are supported in the OpenShift version, meshStack should be able to work with it:
+
+* /api/v1
```
Comment (JohannesRudolph) [inline docs/meshstack.openshift.index.md:22]: d: this should also link to the K8s APIs we use (link to general kubernetes docs). 

d: maybe replace the exact APIs used with a link to the cluster role YAML which lists them as well in the integration section of the docs. "meshStack supports all Kubernetes distributions offering the following API versions"...

File: docs/meshstack.cloudfoundry.index.md [hunk 3]
```diff
@@ -19,6 +19,24 @@ The tenant [replication](./meshcloud.tenant.md) ensures spaces and orgs are crea
 
 ![Cloud Foundry Architecture](assets/cf-architecture.svg)
 
+## CloudFoundry Versions
```
[the remark this answers is not in the document]
Comment (sttomm) [reply docs/meshstack.cloudfoundry.index.md:22]: i know that ;) Just a typo that i fixed now ;)

File: docs/meshstack.openshift.index.md [hunk 4]
```diff
@@ -15,7 +15,12 @@ To enable integration with OpenShift, operators deploy and configure the meshSta
 
 ### OpenShift Versions
 
-meshStack currently supports OpenShift version 3.9+ as either Open-Source (OKD) or OpenShift Enterprise variants.
+meshStack currently supports OpenShift version 3.9+ and 4.x as either Open-Source (OKD) or OpenShift Enterprise variants.
```
[the remark this answers is not in the document]
Comment (sttomm) [reply docs/meshstack.openshift.index.md:18]: actually not. I extended the sentence accordingly.

File: docs/meshstack.openshift.index.md (hunk 2 above)
Comment (sttomm) [reply docs/meshstack.openshift.index.md:22]: I actually linked the cluster roles above. I changed the sentence a bit to make it clearer. It is just that e.g. for namespaces no version is provided of the API. So we have to add it here too.

File: docs/meshstack.openstack.index.md [hunk 5]
```diff
@@ -15,6 +15,25 @@ The backend also uses the JWT token to check the user's permissions. If the perm
...
+
+* `KeystoneV3`
+* `CinderV3`
+* `Neutron v2.0`
+* `Glance V2`
+
+Additionally the following OpenStack APIs are used without specific major version requirements. Still, the specific microversion of these services must support the calls listed here.
+
+* `Nova`: list servers & VM flavors
+* `Swift or Radosgw-Swift`: list Object Storages, via meshPanel also creation of those
```
Comment (Prerna21122112) [inline docs/meshstack.openstack.index.md:32]:
```suggestion
* `Swift or Radosgw-Swift`: list Object Storages, via meshPanel 
```

File: docs/meshstack.openstack.index.md [hunk 6]
```diff
@@ -15,6 +15,25 @@ The backend also uses the JWT token to check the user's permissions. If the perm
...
+* `Heat`: list Heat Stacks for resource check and metering
+* `Designate`: list DNS Zones for resource check and metering
+* `Panko` (optional): List events about resource lifecycle to provide more fine-grained metering
```
Comment (Prerna21122112) [inline docs/meshstack.openstack.index.md:35]:
```suggestion
* `Panko` (optional): lists events about resource lifecycle to provide more fine-grained metering
```

File: docs/meshstack.openstack.index.md [hunk 7]
```diff
@@ -15,6 +15,25 @@ The backend also uses the JWT token to check the user's permissions. If the perm
...
+* `CinderV3`
+* `Neutron v2.0`
+* `Glance V2`
+
+Additionally the following OpenStack APIs are used without specific major version requirements. Still, the specific microversion of these services must support the calls listed here.
+
+* `Nova`: list servers & VM flavors
+* `Swift or Radosgw-Swift`: list Object Storages, via meshPanel also creation of those
+* `Heat`: list Heat Stacks for resource check and metering
+* `Designate`: list DNS Zones for resource check and metering
```
Comment (Prerna21122112) [inline docs/meshstack.openstack.index.md:34]:
```suggestion
* `Designate`: lists DNS Zones for resource check and metering
```

File: docs/meshstack.openstack.index.md [hunk 8]
```diff
@@ -15,6 +15,25 @@ The backend also uses the JWT token to check the user's permissions. If the perm
...
+* `KeystoneV3`
+* `CinderV3`
+* `Neutron v2.0`
+* `Glance V2`
+
+Additionally the following OpenStack APIs are used without specific major version requirements. Still, the specific microversion of these services must support the calls listed here.
+
+* `Nova`: list servers & VM flavors
+* `Swift or Radosgw-Swift`: list Object Storages, via meshPanel also creation of those
+* `Heat`: list Heat Stacks for resource check and metering
```
Comment (Prerna21122112) [inline docs/meshstack.openstack.index.md:33]:
```suggestion
* `Heat`: lists Heat Stacks for resource check and metering
```

**Review by Prerna21122112 — APPROVED** [review]

**Pull request merged**